ironfish/src/wallet/scanner/walletScanner.ts

```diff
@@ -33,7 +33,10 @@ export class WalletScanner {
    * A snapshot of the accounts that have `scanningEnabled` set to true. Used
    * to tell what accounts should be scanned, and from what block.
    */
-  private scanningAccounts = new Array<{ account: Account; scanFrom: HeadValue | null }>()
+  private scanningAccounts = new Array<{
+    account: Account
+    scanFrom: { head: HeadValue | null } | null
```

Comment:
The logic in this module is tricky sometimes. I haven't read the changes yet, but I know I will have a lot of trouble differentiating between `scanFrom === null` and `scanFrom.head === null`. Can we find a way to make the meaning of the values more intuitive? Maybe something like this (if it makes sense):

```
scanFrom: { head: HeadValue } | 'always-scan' | 'something-else'
```

Reply:
Yea I think that's a good idea. you're right that's the main thing I'm trying to differentiate with the change though